File: 2855-maximum-number-of-jumps-to-reach-the-last-index/maximum-number-of-jumps-to-reach-the-last-index.cpp
class Solution {
public:
    int maximumJumps(vector<int>& nums, int target) {
        int n = nums.size();
        vector<int> t(n + 1, INT_MIN);
        t[n - 1] = 0;

        for (int i = n - 2; i >= 0; i--) {
            for (int j = i + 1; j < n; j++) {
                if (abs(nums[j] - nums[i]) <= target && t[j] != INT_MIN) {
                    int temp = 1 + t[j];
                    t[i] = max(t[i], temp);
                }
            }
        }
        return t[0] < 0 ? -1 : t[0];
    }
};

File: 2855-maximum-number-of-jumps-to-reach-the-last-index/maximum-number-of-jumps-to-reach-the-last-index.py
class Solution:
    def maximumJumps(self, nums: List[int], target: int) -> int:
        n = len(nums)
        t = [float('-inf')]*(n+1)
        t[n-1]=0

        for i in range(n-2, -1, -1):
            for j in range(i+1, n):
                if abs(nums[j] - nums[i]) <= target and t[j] != float('-inf'):
                    temp = 1 + t[j]
                    t[i] = max(t[i], temp)

        return -1 if t[0] < 0 else t[0]